### PR TITLE
Make it general

### PR DESCRIPTION
**Pull request by jamescook106**

### CODE REVIEW

**Comment by jamescook106** [comment]
@lgeiger please review the docstrings and the modified test I put in, and then we can swap from draft and get an additional reviewer

File: zookeeper/preprocessing.py [hunk 1]
```diff
@@ -2,15 +2,24 @@
...
-    hence we define to abstract methods, an input (e.g the image for image
-    classification), and an output (e.g the class label for image classification).
-    We also define decoders, which allows use to customize the decoding, and kwargs,
-    which allows one to pass information from the preprocessing to the model (e.g
-    input size.)
+    """An abstract class to be used to define data preprocessing.
+
+    We define a neural network to be a mapping between an input and an output, hence we
+    define to abstract methods, an input (e.g the image for image classification), and
+    an output (e.g the class label for image classification). We also define decoders,
```
Comment (MariaHeuss) [inline zookeeper/preprocessing.py]: This sentence still does not really make sense...

File: zookeeper/preprocessing.py [hunk 2]
```diff
@@ -2,15 +2,24 @@
...
+    which allows use to customize the decoding, and kwargs, which allows one to pass
```
Comment (MariaHeuss) [inline zookeeper/preprocessing.py]: which ALLOW US to customize the decoding, and kwargs. THIS

File: zookeeper/preprocessing.py [hunk 3]
```diff
@@ -21,8 +30,24 @@ def __init__(self, features=None):
 
     @abc.abstractmethod
     def inputs(self, data, training):
+        """A method to define preprocessing for inputs.
+
+        This method needs to be overwritten by all subclasses.
+
+        # Arguments
+        data: A dictionary of type {feature_name: Tensor}
+        training: A optional `boolean` to define if preprocessing is called during training.
```
Comment (MariaHeuss) [inline zookeeper/preprocessing.py]: An optional...

File: zookeeper/preprocessing.py [hunk 4]
```diff
@@ -21,8 +30,24 @@ def __init__(self, features=None):
...
     def outputs(self, data, training):
+        """A method to define preprocessing for outputs.
+
+        This method needs to be overwritten by all subclasses.
+
+        # Arguments
+        data: A dictionary of type {feature_name: Tensor}
+        training: A optional `boolean` to define if preprocessing is called during training.
+        """
         raise NotImplementedError("Must be implemented in subclasses.")
```
Comment (MariaHeuss) [inline zookeeper/preprocessing.py]: An optional...

**Comment by jamescook106** [comment]
@MariaHeuss I have updated the doc strings as per your comments

**Pull request merged**